libraries/AP_Camera/AP_Camera.h

```diff
@@ -68,6 +69,9 @@ class AP_Camera {
 
     static const struct AP_Param::GroupInfo        var_info[];
 
+    // update flight control mode. The control mode is vehicle type specific
+    void update_control_mode(uint8_t mode) { _control_mode = mode; }
```

Comment:
Why does this need to know the mode? It looks like it only needs to know if it is in Auto?

Reply:
I did not aound another way to detect Auto, "AUTO" is not defined so that AP_Camera can see it.

Reply:
Precisely, Auto is only known to vehicles but you are inserting that knowledge into the library. You can just change this method to `set_auto_mode(bool enable)` and then in vehicles do something like `camera.set_auto_mode(mode == AUTO);`

Reply:
Thanks for all input, will rework it tomorrow :)